### PR TITLE
Create an event for m.room.server_acl

### PR DESCRIPTION
**Pull request by nvrWhere**
Title

### CODE REVIEW

**Review by TobiasFella — DISMISSED** [review]

**Review by CarlSchwan — DISMISSED** [review]

File: Quotient/events/serveraclevent.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+namespace Quotient {
+class QUOTIENT_API ServerAclEvent : public StateEvent {
+public:
+    QUO_EVENT(ServerAclEvent, "m.room.server_acl")
+
+    using StateEvent::StateEvent;
+
+    QStringList allow() const;
+    bool allowIpLiterals() const;
+    QStringList deny() const;
```
Comment (KitsuneRal) [inline Quotient/events/serveraclevent.h]:
```suggestion
    QUO_CONTENT_GETTER(QStringList, allow)
    QUO_CONTENT_GETTER(bool, allowIpLiterals)
    QUO_CONTENT_GETTER(QStringList, deny)
```
And you don't need the .cpp file at all; but if (see the headline comment) you keep this file, please add it to `CMakeLists.txt`.

File: Quotient/events/serveraclevent.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#pragma once
+
+#include "stateevent.h"
+
+namespace Quotient {
+class QUOTIENT_API ServerAclEvent : public StateEvent {
+public:
+    QUO_EVENT(ServerAclEvent, "m.room.server_acl")
```
Comment (KitsuneRal) [inline Quotient/events/serveraclevent.h]:
```suggestion
class QUOTIENT_API RoomServerAclEvent : public StateEvent {
public:
    QUO_EVENT(RoomServerAclEvent, "m.room.server_acl")
```
To follow the naming convention for (most of) other room events. Whether to rename the file is up to you but if I were to keep it (see above) I would rename it too.

**Review by KitsuneRal — CHANGES_REQUESTED** [review]
See suggestions below; also, since the event class is really small, it would probably make sense to just stuff it into `events/simple_state_events.h`.

File: Quotient/events/serveraclevent.h (hunk 1 above)
Comment (nvrWhere) [reply Quotient/events/serveraclevent.h]: Happy to move into `simplestateevents.h` and get rid of the rest, didn't know this was an option.

File: Quotient/events/serveraclevent.h (hunk 2 above)
Comment (nvrWhere) [reply Quotient/events/serveraclevent.h]: Updated

**Review by KitsuneRal — APPROVED** [review]
Perfect, thanks!

**Pull request merged**